### PR TITLE
Refactor Webserver to make it more testable; Added mock test cases fo…

### PR DESCRIPTION
**Pull request by ghost**
…r Webserver class, gmock working. Need to clean up the Makefile

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  NginxConfig config;
+
+  Webserver server(&parser,&config);
+
+  if(!server.run_server(argv[1])){
+  	return 1;
+  }
+  
+  return 0;
+}
```
Comment (avikbmohan) [inline main.cc]: Might want to add ending newline, but this is unlikely to cause issues (similarly in config_parser.h below)

File: nginx-configparser/config_parser.h [hunk 2]
```diff
@@ -57,3 +59,4 @@ class NginxConfigParser {
 
   TokenType ParseToken(std::istream* input, std::string* value);
 };
+#endif
```
Comment (richardmin) [inline nginx-configparser/config_parser.h]: Append a new line here.

File: webserver.cc [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
-  if (argc != 2) {
-    printf("Usage: ./webserver <path to config file>\n");
-    return 1;
-  }
-
-  NginxConfigParser config_parser;
-  NginxConfig config;
-  
-  if (!config_parser.Parse(argv[1], &config)) {
+bool Webserver::parse_config(const char* file_name){
```
Comment (richardmin) [inline webserver.cc]: Is this code redundant? Why does the nginx-configparser have a wrapper here?

File: webserver.cc [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
-    return 1;
+    return false;
   }
+  return true;
 
+}
 
-  std::string port_str = "";
-  if (!config.find(port_str)) {
+bool Webserver::get_port_from_config(std::string& s){
```
Comment (richardmin) [inline webserver.cc]: Single user functions like this shouldn't exist. This adds no logic (for testing, as that should be covered by nginx-configparser tests)

**Review by richardmin — CHANGES_REQUESTED** [review]

File: main.cc (hunk 1 above)
Comment (ghost) [reply main.cc]: ok

File: nginx-configparser/config_parser.h (hunk 2 above)
Comment (ghost) [reply nginx-configparser/config_parser.h]: sure

File: webserver.cc (hunk 3 above)
Comment (ghost) [reply webserver.cc]: I can just merge them into one function

File: webserver.cc (hunk 4 above)
Comment (ghost) [reply webserver.cc]: same as above

**Comment by richardmin** [comment]
Ship it

**Review by richardmin — APPROVED** [review]

**Pull request merged**